### PR TITLE
Slip days logic

### PR DESCRIPTION
**Pull request by meling**
This mainly implements the slip days logic on the backend, and should also be usable from the frontend through the `Enrollment` message type. It should be possible to implement a frontend function similar to the `TotalSlipDays()` or `RemainingSlipDays()` methods.

The functionality should work (I hope). However, it has not been hooked up with the database yet.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -33,7 +34,7 @@ ui:
 
 proto:
 	@echo Compiling Autograders proto definitions
-	@cd ag; protoc -I=. -I=$(GOPATH)/src -I=$(GOPATH)/src/github.com/gogo/protobuf/protobuf --gogofast_out=plugins=grpc,\
+	@cd ag; protoc -I=. -I=$(pbpath) --gogofast_out=plugins=grpc,\
```
Comment (meling) [inline Makefile:37]: Does this work for you?

File: ag/ag.proto [hunk 2]
```diff
@@ -1,6 +1,8 @@
 syntax = "proto3";
 
-import "github.com/gogo/protobuf/gogoproto/gogo.proto";
+//TODO(meling) workaround for Go module system
+import "gogoproto/gogo.proto";
```
Comment (meling) [inline ag/ag.proto:4]: And this also causes no problem?

File: Makefile (hunk 1 above)
Comment (0xf8f8ff) [reply Makefile:37]: Yes, flawlessly.

File: ag/ag.proto (hunk 2 above)
Comment (0xf8f8ff) [reply ag/ag.proto:4]: Well, somehow my VSCode does not really get the idea of imports in `.proto` files and throws the same error for both paths, while the compiler has no problems with either. So, basically, yes, no problems.

File: ag/enrollment.go [hunk 3]
```diff
@@ -0,0 +1,68 @@
+package ag
+
+import (
+	"fmt"
+	"time"
+)
+
+// TODO(meling) must also handle groups; if group assignment is late; withdraw slip days per group member.
+// TODO(meling) also need to propogate slip day information to frontend to show to user
```
Comment (meling) [inline ag/enrollment.go:9]: If these are handled now, we can remove them.

**Pull request merged**